### PR TITLE
Homepage api

### PR DESCRIPTION
**Pull request by ayanjamil**

### CODE REVIEW

File: screens/Home.js [hunk 1]
```diff
@@ -5,26 +5,43 @@ import {
...
 
-const { width, height } = Dimensions.get("screen");
-
-const thumbMeasure = (width - 48 - 32) / 3;
-const loading = false;
+const { width } = Dimensions.get("screen");
 
 const Home = () => {
+  const [result, setResult] = useState(null);
+  const [loading, setLoading] = useState(false);
```
Comment (udaybansal19) [inline screens/Home.js]: set initial state as true as the app should open to the explore page loading

File: constants/homepageData.js
Comment (udaybansal19) [inline constants/homepageData.js:1]: what is the use of this file?

File: screens/Home.js [hunk 2]
```diff
@@ -5,26 +5,43 @@ import {
...
+        "We're unable to fetch data right now. Please try again later."
+      );
+    } finally {
+      setLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    getData(url);
+  }, [url]);
```
Comment (udaybansal19) [inline screens/Home.js]: we don't need to pass url to the function and in the dependency array. the url is constant

File: screens/Home.js [hunk 3]
```diff
@@ -5,26 +5,43 @@ import {
...
+  const getData = useCallback(async (feedDataURL) => {
+    setLoading(true);
+    try {
+      const response = await fetch(feedDataURL);
+      const searchResultsData = await response.json();
+      setResult(searchResultsData.data);
+    } catch (error) {
+      Alert.alert(
+        "Error",
+        "We're unable to fetch data right now. Please try again later."
```
Comment (udaybansal19) [inline screens/Home.js]: Change the message
"Sorry we couldn't find anything for you at the moment :(("

File: package.json [hunk 4]
```diff
@@ -22,6 +22,7 @@
     "@react-navigation/native": "^6.0.10",
     "@react-navigation/stack": "^6.2.1",
     "@use-expo/font": "^2.0.0",
+    "axios": "^1.6.7",
```
Comment (udaybansal19) [inline package.json]: what library did we use for the search api? can't we reuse that here?

File: context/AuthContext.js [hunk 5]
```diff
@@ -1,19 +1,49 @@
...
-    setUserToken(token);
+
+  const authFinish = (userUid) => {
+    setUserToken(userUid);
     setIsLoading(false);
-    console.log(token);
+    storeUserToken(userUid);
   };
+
+  const retrieveData = async () => {
```
Comment (ayanjamil) [inline context/AuthContext.js:14]: earlier the app was not able to store the userToken so every time the app reloaded the user was prompted to go through the onboarding part again, used asyncstorage to sore the userToken value

File: context/AuthContext.js [hunk 6]
```diff
@@ -1,19 +1,49 @@
-import React, { children, createContext, useState } from "react";
+import React, { createContext, useState, useEffect } from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
```
Comment (ayanjamil) [inline context/AuthContext.js:2]: earlier the app was not able to store  userToken locally , so every time the app reloaded the user was prompted to go through the onboarding part again, used asyncstorage to save the userToken value

File: screens/SignUp.js [hunk 7]
```diff
@@ -14,8 +14,8 @@ const { height, width } = Dimensions.get("screen");
 import { useNavigation } from "@react-navigation/native";
```
Comment (ayanjamil) [inline screens/SignUp.js:14]: In Firestore documents were getting stored to a newly generated doc ref id but it should have been with the userUid which was generated when the user passed the onboarding pages so used a different function to do that

File: screens/SignUp.js [hunk 8]
```diff
@@ -14,8 +14,8 @@ const { height, width } = Dimensions.get("screen");
...
 
 import { createUserWithEmailAndPassword } from "firebase/auth";
-import { auth, db } from "../firebase";
-import { collection, addDoc } from "firebase/firestore";
+import { auth, db } from "../config/firebase";
+import { doc, setDoc } from "firebase/firestore";
```
Comment (ayanjamil) [inline screens/SignUp.js:18]: In Firestore documents were getting stored to a newly generated doc ref id but it should have been with the userUid which was generated when the user passed the onboarding pages so used a different function to do that

File: context/AuthContext.js (hunk 5 above)
Comment (udaybansal19) [inline context/AuthContext.js]: add this in useCallback

File: context/AuthContext.js [hunk 9]
```diff
@@ -1,19 +1,49 @@
...
+      }
+    } catch (error) {
+      console.error(error);
+    }
+    return null;
+  };
+
+  useEffect(() => {
+    retrieveData().then((token) => {
+      setUserToken(token);
```
Comment (udaybansal19) [inline context/AuthContext.js]: we don't use .then syntax anymore with ES20 syntax for javascript. we use async/await syntax instead

File: context/AuthContext.js [hunk 10]
```diff
@@ -1,19 +1,49 @@
...
     setIsLoading(false);
-    console.log(token);
+    storeUserToken(userUid);
   };
+
+  const retrieveData = async () => {
+    try {
+      const value = await AsyncStorage.getItem("userToken");
+      if (value !== null) {
+        return value;
```
Comment (udaybansal19) [inline context/AuthContext.js]: you can directly set the state here for setUserToken(value)

File: context/AuthContext.js [hunk 11]
```diff
@@ -1,19 +1,49 @@
...
+    return null;
+  };
+
+  useEffect(() => {
+    retrieveData().then((token) => {
+      setUserToken(token);
+    });
+  }, []);
+
   const logout = () => {
```
Comment (udaybansal19) [inline context/AuthContext.js]: while logging out remove token from async storage as well or on next load the user would get logged in again

File: context/AuthContext.js [hunk 12]
```diff
@@ -1,19 +1,49 @@
...
     setUserToken(null);
     setIsLoading(false);
   };
+
+  const storeUserToken = async (userUid) => {
```
Comment (udaybansal19) [inline context/AuthContext.js]: wrap in useCallback

File: screens/SignIn.js [hunk 13]
```diff
@@ -22,8 +22,8 @@ import { AuthContext } from "../context/AuthContext";
 const SignIn = (props) => {
   const { authFinish } = useContext(AuthContext);
   const navigation = useNavigation();
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("ayanjamil00@gmail.com");
```
Comment (udaybansal19) [inline screens/SignIn.js]: remove creds from here

File: context/AuthContext.js [hunk 14]
```diff
@@ -1,19 +1,57 @@
...
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await retrieveData();
+    };
+    fetchData();
```
Comment (udaybansal19) [inline context/AuthContext.js]: you can just call retreiveData() directly here 😅 
No need for fetchData function

File: context/AuthContext.js [hunk 15]
```diff
@@ -1,19 +1,57 @@
...
+import AsyncStorage from "@react-native-async-storage/async-storage";
 export const AuthContext = createContext();
 export const AuthProvider = ({ children }) => {
-  const [isLoding, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
   const [userToken, setUserToken] = useState(null);
-  const authFinish = (token) => {
-    setUserToken(token);
+
+  const authFinish = (userUid) => {
```
Comment (udaybansal19) [inline context/AuthContext.js]: make it a general practice to wrap all functions in useCallback

File: navigation/AppNav.js [hunk 16]
```diff
@@ -9,6 +9,7 @@ import { argonTheme } from "../constants";
 import { AuthContext } from "../context/AuthContext";
 const AppNav = () => {
   const { isLoading, userToken } = useContext(AuthContext);
+  //userUid is passed as userToken in AuthContext
```
Comment (udaybansal19) [inline navigation/AppNav.js]: what does this mean?

File: context/AuthContext.js [hunk 17]
```diff
@@ -1,19 +1,57 @@
...
+    removeData("userToken");
   };
+
+  const storeUserToken = useCallback(async (userUid) => {
+    try {
+      await AsyncStorage.setItem("userToken", userUid);
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
```
Comment (udaybansal19) [inline context/AuthContext.js:41]: take care of indentation after this line. there is no line break

**Review by udaybansal19 — APPROVED** [review]

**Pull request merged**